main.py

```diff
@@ -392,4 +388,70 @@ def check(msg):
             await ctx.send(file=discord.File('assets/pap.jpg'))
 
 
+# @bot.command()
+# async def guess(ctx):
+#     embed = discord.Embed(
+#         title="Guessing Game 🥸👀",
+#         description="Try and guess the number I'm thinking.🤔\nYou only get 3 chances to guess the correct number."
+#                     "\nDkn what to put yet!!!",
+#
+#         color=discord.Colour.red()
+#     )
+#     tries = 3
+#     number = random.randrange(50)                                   STILL  UNDER WORK
+#     amount_tries = 0
+#     await ctx.send(embed=embed)
+#
+#     def check(msg):
+#         return msg.author == ctx.author and msg.channel == ctx.channel and msg.content.lower()
+#
+#     user_choice = await bot.wait_for('message', check=check)
+#     while tries == 3:
+#         user_choice = await bot.wait_for('message', check=check)
+#         user_choice = user_choice.content
+#         print("hello")
+#         if user_choice == number:
+#             await ctx.send("Congrats you found the number")
+
+
+
+@bot.command()
+async def grps(ctx):
+    group_size = 4
+    min_group_size = 2
+
+    embed = discord.Embed(
+        title="Group generator",
+        description="React to the message to get assigned in a group",
+        color=discord.Colour.yellow()
+    )
+    reaction = '👍'
+    message = await ctx.send(embed=embed)
+    await message.add_reaction(reaction)
+
+    def check(reaction, user):
+        return user != bot.user and str(reaction.emoji) == reaction and reaction.message.id == message.id
+
+    while True:
+        reaction, user = await bot.wait_for('reaction_add', check=check)
+        members = await reaction.users().flatten()
+        # Remove the bot from the list of members
+        members.remove(bot.user)
+        # If there are not enough members for a group, do nothing
+        if len(members) < min_group_size:
+            continue
+        # If there are enough members, but not enough for a full group, make a group with all of them
+        elif len(members) < group_size:
+            group = ", ".join(member.mention for member in members)
+            await user.send(f"You are in a group with {group}")
+        # If there are enough members for multiple groups, split them into groups
+        else:
+            # Shuffle the list of members to assign them randomly to groups
+            random.shuffle(members)
+            groups = [members[i:i + group_size] for i in range(0, len(members), group_size)]
+            for i, group in enumerate(groups):
+                group_str = ", ".join(member.mention for member in group)
+                await group[0].send(f"You are in group {i + 1} with {group_str}")
+
+
```

Comment:
Consider the following refactor:

1. We send command.
2. Bot posts message and waits for 3 minutes.
3. After 3 minutes, bot adds all users who reacted to a list
4. Split up the list into smaller lists (groups)
5. Send message to members in each group